### PR TITLE
Add support for the new artifact format

### PR DESCRIPTION
**Pull request by sihil**
This provides support (by default) for the new artifact format created in Riff-Raff (https://github.com/guardian/riff-raff/pull/311).

An app can go back to the legacy format by using `legacySettings` if they so desire.

### CODE REVIEW

File: src/main/scala/com/gu/riffraff/artifact/RiffRaffArtifact.scala [hunk 1]
```diff
@@ -45,10 +45,10 @@ object RiffRaffArtifact extends AutoPlugin {
     lazy val riffRaffUploadArtifactBucket = settingKey[Option[String]]("Bucket to upload artifacts to")
     lazy val riffRaffUploadManifestBucket = settingKey[Option[String]]("Bucket to upload manifest to")
 
-    lazy val riffRaffNotifyTeamcity = settingKey[Boolean]("Report artifacts to Teamcity if true")
+    lazy val riffRaffNotify = taskKey[Unit]("Task to notify teamcity")
```
Comment (philwills) [inline src/main/scala/com/gu/riffraff/artifact/RiffRaffArtifact.scala]: I like making this a task, but I'm not terribly keen on the name.

**Comment by philwills** [comment]
Travis doesn't seem to think it compiles.

File: src/main/scala/com/gu/riffraff/artifact/RiffRaffArtifact.scala (hunk 1 above)
Comment (sihil) [inline src/main/scala/com/gu/riffraff/artifact/RiffRaffArtifact.scala]: Do you think I could eliminate the setting?

Comment (philwills) [inline src/main/scala/com/gu/riffraff/artifact/RiffRaffArtifact.scala]: Ooh, good idea. That makes sense to me. I'd give it a slightly bigger version bump and email dig.dev, but that seems a better solution.

Comment (sihil) [inline src/main/scala/com/gu/riffraff/artifact/RiffRaffArtifact.scala]: OK - done. The task has the name that the setting previously did.

**Comment by philwills** [comment]
👍

**Comment by sihil** [comment]
I advise against releasing this until https://github.com/guardian/riff-raff/pull/311 has been released and we're confident that it will not be rolled back.

**Comment by philwills** [comment]
guardian/riff-raff#311 has been out for a week now, so let's give this a go

**Pull request merged**